### ISSUE TITLE
aws-appsync: invalid service role creation

### ISSUE THREAD

**Issue by mtliendo** [aws/aws-cdk#23035]
### Describe the bug

When attaching a DynamoDB datasource to an API, the generated service role does not include the correct servicerole name
![image](https://user-images.githubusercontent.com/5106417/203350904-74809680-bb8f-4840-95e8-a1171a2bbe22.png)

(it's missing the `.amazonaws` part)

### Expected Behavior

I would expect the correct service role to be created.

### Current Behavior

On version 2.51.1 I created [the following stack](https://github.com/focusOtter/cdk-fullstack-kitchen-sink) and deployed the application but it failed on the deploying the API. After inspecting the output, I found that [the datasource](https://github.com/focusOtter/cdk-fullstack-kitchen-sink/blob/main/lib/APIStack.ts#L49-L52) was not given the correct datasource.

Reverting back to v2.50.0 had the correct service role
![image](https://user-images.githubusercontent.com/5106417/203352094-cfd95f35-877e-4609-8c72-609803c11de8.png)

### Reproduction Steps

- clone the repo
- update to the latest version
- deploy
- view error
- remove node_modules directory and package-lock.json file
- revert to v2.50.0
- deploy
- view successful deploy

### Possible Solution

Appending the correct principal

### CDK CLI Version

2.51.1

### Framework Version

CDK

### Node.js Version

16

### OS

mac

### Language

Typescript

**Comment by jpignata** [comment]
Is this related to https://github.com/aws/aws-cdk/commit/65d8e3d7814ac69d6cfa2bfd0d4219817028a14b? Do the references to the service principal need to be updated? The commit indicates this change was behind a feature flag.

**Comment by jpignata** [comment]
It appears this reference (https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-appsync/lib/data-source.ts#L116) needs to be updated.

**Comment by peterwoodworth** [comment]
I'm finding this new incorrect behavior is only being applied with the feature flag enabled. You should be able to disable the feature flag for now while we get this fixed up. Thanks for reporting!

**Comment by mtliendo** [comment]
Ah yea, that seems to be the issue. On new CDK projects, the default state of the flag is set to `true`.

**Comment by ryparker** [comment]
Thanks to @robertd for the fix, it has been merged and will be included in the next release shortly after re:invent.

Here’s a workaround for the interim:

Disable the [`@aws-cdk/aws-iam:standardizedServicePrincipals`](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/cx-api/FEATURE_FLAGS.md#aws-cdkaws-iamstandardizedserviceprincipals) feature flag in your `cdk.json`.

`cdk.json`:
```json
{
  "context": {
    "@aws-cdk/aws-iam:standardizedServicePrincipals": false
  }
}
```
_[Feature flag docs](https://docs.aws.amazon.com/cdk/v2/guide/featureflags.html)_

**Comment by ryparker** [comment]
Patched in `aws-cdk-lib/aws-appsync` version [2.52.1-alpha.0](https://github.com/aws/aws-cdk/releases/tag/v2.52.1).**
